internal/analytics/tilt_analytics.go

```diff
@@ -0,0 +1,72 @@
+package analytics
+
+import (
+	"time"
+
+	"github.com/windmilleng/wmclient/pkg/analytics"
+)
+
+// An Analytics that:
+// 1. Has `IncrIfUnopted` to report anonymous metrics only for users who have not opted in/out (or the choice that they
+//    did opt in/out).
+// 2. Ignores all other calls from users who have not opted in.
+// 3. Allows opting in/out at runtime.
+type TiltAnalytics struct {
+	opt        analytics.Opt
+	persistOpt optPersister
+	a          analytics.Analytics
+}
+
+type optPersister func(analytics.Opt) error
+
+func NewTiltAnalytics(opt analytics.Opt, setOpt optPersister, analytics analytics.Analytics) *TiltAnalytics {
+	return &TiltAnalytics{opt, setOpt, analytics}
+}
+
+func (ta *TiltAnalytics) Count(name string, tags map[string]string, n int) {
+	if ta.opt == analytics.OptIn {
+		ta.a.Count(name, tags, n)
+	}
+}
+
+func (ta *TiltAnalytics) Incr(name string, tags map[string]string) {
+	if ta.opt == analytics.OptIn {
+		ta.a.Incr(name, tags)
+	}
+}
+
+func (ta *TiltAnalytics) IncrIfUnopted(name string, tags map[string]string) {
+	if ta.opt == analytics.OptDefault {
+		ta.a.IncrAnonymous(name, tags)
+	}
+}
+
+func (ta *TiltAnalytics) IncrAnonymous(name string, tags map[string]string) {
+	if ta.opt == analytics.OptIn {
```

Comment:
wait, I don't think you want this check here 🤔

Reply:
Maybe? It was intentional. I could go either way, and I think it's weird either way. I was going to explain my thinking here, but added it as a comment (the best way to paper over poor design) instead. That isn't a doubling down on this choice, and I'm open to being convinced otherwise - I've just seen (and written!) a number of explanations in PR comments that would have been better in the code itself.